Source/pyboy/pyboy.py

```diff
@@ -119,11 +126,25 @@ def stop(self, save=True):
         if self.profiling:
             print("Profiling report:")
             from operator import itemgetter
-            names = [CPU_COMMANDS[n] if n < 0x100 else CPU_COMMANDS_EXT[n-0x100] for n in range(0x200)]
+            names = [CPU_COMMANDS[n] if n < 0x100 else CPU_COMMANDS_EXT[n - 0x100] for n in range(0x200)]
             for hits, n, name in sorted(
                     filter(itemgetter(0), zip(self.mb.cpu.hitRate, range(0x200), names)), reverse=True):
                 print("%3x %16s %s" % (n, name, hits))
 
+    def step(self, action):
+        # === Internal Process to move one step forward ===
+        self.mb.buttonevent(action)
+        if not self.paused:
+            self.mb.tickframe() # TODO: I don't know what this does...
+        self.window.update_display() # update the screen
+        # === Internal Process done ===
+
+        observation = self.get_obs()
+        reward = -1.0  # TODO: think the reward design
+        done = False # TODO: when to done??
+        info = {}
+        return observation, reward, done, info
```

Comment:
If OpenAI expects a specific format for the step function, I would prefer to have it in a separate wrapper class/file.